community/modules/file-system/gke-persistent-volume/main.tf

```diff
@@ -18,3 +18,71 @@ locals {
   # This label allows for billing report tracking based on module.
   labels = merge(var.labels, { ghpc_module = "gke-persistent-volume" })
 }
+
+locals {
+  split_filestore_id   = var.filestore_id != null ? split("/", var.filestore_id) : [null, null, null, null, null, null]
+  location             = local.split_filestore_id[3]
+  filestore_name       = local.split_filestore_id[5]
+  filestore_share_name = trimprefix(var.network_storage.remote_mount, "/")
+
+  pv_name  = "${local.filestore_name}-pv"
```

Comment:
It appears to me that you are doing nothing to address `var.filestore_id == null` here. Do you have a plan? It seems to me that, in this initial commit, `var.filestore_id == null` probably shouldn't be allowed.

My plan forward (assuming the intent is to also support Cloud Storage buckets, etc.) would be:

1. Refactor each type of storage as a submodule of this module (directly in the filesystem path)
2. Instantiate each submodule only if the matching input variable is defined
3. Merge the outputs of each submodule into a set of PVs, PVCs
4. Pluralize the output values

Reply:
Good suggestion using submodules. I had more or less considered doing this but all in locals, submodules may be cleaner. Do you want to see these changes as part of this PR or can it be follow on. 

I could imagine, this pr, then separate pr to move to submodule, then pr adding gcs.

Reply:
See https://github.com/GoogleCloudPlatform/hpc-toolkit/pull/1442#discussion_r1230123258 for the "right now" suggestions but I think you'll find the suggestion potentially simplifying.